### DISCUSSION TITLE
Wazuh vulnerability detection showing really old (often 2006) CVE's being referenced incorrectly

### DISCUSSION THREAD

**Discussion by voidedsquirrel** [wazuh/wazuh#24307] [General]
Trying to make use of this new improved vulnerability detection functionality, but it's not really giving me a useful dashboard. I'm seeing things like FireFox version 127.0 with CVE's from 2006 - 2014! 

Similar with other OS and applications.

Anyone know why this is and what I can do to fix it?

Thanks

**Comment by pradoh97** [comment]
Same here! I'm attaching one of the several I'm receiving just for reference note that the vulnerability was found for office 2000-2003 but the installed product is office 2021 on its latest version:

Rule: 23505 fired (level 10) -> "CVE-2006-1540 affects Microsoft Office Profesional Plus 2021 - es-es"
Portion of the log(s):
        Timestamp: "2024-07-03T16:35:24.257-0300"
        Rule
            Level: 10
            Description: "CVE-2006-1540 affects Microsoft Office Profesional Plus 2021 - es-es"
            Id: "23505"
            Firedtimes: 2
            Groups: "vulnerability-detector"
            Gdpr: "IV_35.7.d"
            Pci_dss: "11.2.1" "11.2.3"
            Tsc: "CC7.1" "CC7.2"
        Manager
            Name: "wazuh"
        Data
            Vulnerability
                Assigner: "mitre"
                Cve: "CVE-2006-1540"
                Cvss
                    Cvss2
                        Base_score: "9.300000"
                        Vector
                Cwe_reference: "CWE-94"
                Enumeration: "CVE"
                Package
                    Architecture: "x86_64"
                    Condition: "Package default status"
                    Name: "Microsoft Office Profesional Plus 2021 - es-es"
                    Source: " "
                    Version: "16.0.17726.20126"
                Published: "2006-03-30T11:02:00Z"
                Rationale: "MSO.DLL in Microsoft Office 2000, Office XP (2002), and Office 2003 allows user-assisted attackers to cause a denial of service and execute arbitrary code via multiple attack vectors, as originally demonstrated using a crafted document record with a malformed string, as demonstrated by replacing a certain \"01 00 00 00\" byte sequence with an \"FF FF FF FF\" byte sequence, possibly causing an invalid array index, in (1) an Excel .xls document, which triggers an access violation in ole32.dll; (2) an Excel .xlw document, which triggers an access violation in excel.exe; (3) a Word document, which triggers an access violation in mso.dll in winword.exe; and (4) a PowerPoint document, which triggers an access violation in powerpnt.txt. NOTE: after the initial disclosure, this issue was demonstrated by triggering an integer overflow using an inconsistent size for a Unicode \"Sheet Name\" string."

**Reply by Damian-Mangold** [reply]
Hi @pradoh97. Thank you for the report. As you mentioned, the CVE has been identified as a false positive. This case has been acknowledged and is currently undergoing the sanitization process. [Here](https://github.com/wazuh/wazuh/issues/24178#issuecomment-2179352568) is the tracking issue for your reference.

**Comment by Damian-Mangold** [comment]
Hi @voidedsquirrel. Thank you for the report. I understand that your message is from an earlier time, but do you have the list of CVEs that you consider false positives?

Could you please share the following information with me to enable a more detailed investigation?

### 1. **[OS Information](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_os_info), [Package Inventory](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_packages_info), and [Hotfix Information](https://documentation.wazuh.com/current/user-manual/api/reference.html#operation/api.controllers.syscollector_controller.get_packages_info)**

<details><summary>Details</summary>
<p>

To collect these details, please follow these steps:

1. **Access the Wazuh Dashboard**: 
   - Navigate to the left-hand menu and select **"Server management"** -> **"Dev Tools"**.
![image](https://github.com/user-attachments/assets/33f3773d-41f0-4d4d-9932-08749786092f)

2. **Query the Relevant Data**: 
   - In the **"Dev Tools"** section, you will find a query input field on the left side. You will need the agent ID to perform the query.
   - Once you have entered the query, click the green play button. The results will be displayed on the right side.
![image](https://github.com/user-attachments/assets/c6cdec78-8fd8-4919-9c7e-a9227ec58c92)

   Please execute the following queries:

   - `GET /syscollector/<agentID>/os` - Retrieves the operating system information.
   - `GET /syscollector/<agentID>/packages` - Retrieves the list of installed packages.
   - `GET /syscollector/<agentID>/hotfixes` - Retrieves the installed hotfixes (for Windows OS).

</p>
</details> 



### 2. **Vulnerabilities Inventory**

<details><summary>Details</summary>
<p>

To extract the vulnerability information, please follow these steps:

1. **Access the Vulnerability Detection Module**:
   - In the Wazuh Dashboard, go to the left-hand menu and select **"Threat intelligence"** -> **"Vulnerability Detection"**.
![image](https://github.com/user-attachments/assets/6878b52e-4adb-4766-be77-ac945e70ec02)

2. **Export the Inventory**:
   - Once inside the **"Vulnerability Detection"** module, navigate to the **"Inventory"** tab.
   - Use the **"Export Formatted"** option to download the inventory data.
![image](https://github.com/user-attachments/assets/fa6d8595-aeac-4b60-966d-42a8e29c7db3)

</p>
</details>

**Comment by sebasfalcone** [comment]
Inactive thread, please re-open it if necessary
